engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/mutable/ArrayLengthNode.java

```diff
@@ -0,0 +1,24 @@
+package org.enso.interpreter.node.expression.builtin.mutable;
+
+import com.oracle.truffle.api.CompilerDirectives;
+import com.oracle.truffle.api.interop.InteropLibrary;
+import com.oracle.truffle.api.interop.UnsupportedMessageException;
+import com.oracle.truffle.api.nodes.Node;
+import org.enso.interpreter.dsl.BuiltinMethod;
+import org.enso.interpreter.runtime.Context;
+import org.enso.interpreter.runtime.error.PanicException;
+
+@BuiltinMethod(type = "Array", name = "length", description = "Length of polyglot array")
+public class ArrayLengthNode extends Node {
+  @Child InteropLibrary iop = InteropLibrary.getFactory().createDispatched(3);
+
+  long execute(Object self) {
+    try {
+      return iop.getArraySize(self);
+    } catch (UnsupportedMessageException ex) {
```

Comment:
If I understand correctly, this is an impossible state? Probably should signal this to the user by using something like `IllegalStateException`

Reply:
It is interop. Nothing is impossible. We don't know into what implementation we are calling.

`IllegalStateException` kills the engine. When it happens in a test suite, the whole execution ends. Is there any benefit in using it rather than reasonable, recoverable `PanicException`?

Reply:
Plain panic as in case of 4d9dfb440b7 would also work, I guess.

Reply:
This whole class can be replaced by just
```
@@ -116,7 +118,9 @@ public final class Array implements TruffleObject {
    * @return the size of this array
    */
   @ExportMessage
-  long getArraySize() {
+  @Builtin.Method(name = "length", description = "Returns the length of this Array.")
+  @Builtin.Specialize
+  public final long getArraySize() {
```
in Array

Reply:
@JaroslavTulach yes, the benefit is that the user won't recover from bugs in GraalVM. I.e. if this ever happens I want to know and I want to fix it in the interpreter, not just have users deal with panics that come from the interpreter (or: a Truffle language) being broken